src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,534 @@
+ *    Hail falls in a two-space radius around her units,
+ *      changing the weather in those tiles to snow,
+ *      damaging enemies for up to 2HP, and
+ *      destroying any forests (reducing them to grass).
+ *
+ * Likes: Sleigh rides, Hot Chocolate
+ * Dislikes: Jungles, Large Cats
+ */
+public class CommanderAve extends Commander
+{
+  public static final int SNOW_THRESHOLD = 100; // Big numbers for integer math.
+  public static final int SNOW_PER_TURN = 400;
+  public static final int SNOW_MELT_RATE = 100;
+  private int MAX_SNOW_SPREAD_RANGE = 5;
+  private CitySnowifier snowifier;
+
+  private static final CommanderInfo coInfo = new CommanderInfo("Ave", new instantiator());  
+  private static class instantiator implements COMaker
+  {
+    @Override
+    public Commander create()
+    {
+      return new CommanderAve();
+    }
+  }
+
+  private int[][] snowMap;
+
+  public CommanderAve()
+  {
+    super(coInfo);
+
+    // Ave's units are fine in the snow, but not in the trees.
+    for( UnitModel um : unitModels )
+    {
+      for( TerrainType terrain : TerrainType.TerrainTypeList )
+      {
+        um.propulsion.setMoveCost(Weathers.SNOW, terrain, um.propulsion.getMoveCost(Weathers.CLEAR, terrain));
+      }
+      for( Weathers weather : Weathers.values() )
+      {
+        um.propulsion.setMoveCost(weather, TerrainType.FOREST, um.propulsion.getMoveCost(weather, TerrainType.FOREST)+1);
+      }
+    }
+
+    addCommanderAbility(new GlacioAbility(this));
+    snowifier = new CitySnowifier(this);
+    GameEventListener.registerEventListener(snowifier);
+  }
+
+  @Override
+  public GameEventQueue initTurn(GameMap gameMap)
+  {
+    GameEventQueue returnEvents = super.initTurn(gameMap);
+
+    // Initialize our snow tracker if needed.
+    if( null == snowMap )
+    {
+      System.out.println("First time init: adding one snow");
+      // Start by initializing owned properties to SNOW_THRESHOLD
+      snowMap = new int[gameMap.mapWidth][gameMap.mapHeight];
+      addSnow(SNOW_THRESHOLD, gameMap, returnEvents);
+    }
+
+    // Add snow to all buildings we own.
+    addSnow(SNOW_PER_TURN, gameMap, returnEvents); // Enough to spread a bit each way.
+    relevelSnow(gameMap, returnEvents); // Enough to spread a bit each way.
+
+    if(snowLoggingEnabled) log("[Commander Ave.initTurn] Snow Map: \n" + getSnowMapAsString());
+    return returnEvents;
+  }
+
+  /** Ave's units take less cover from forests. */
+  @Override
+  public void applyCombatModifiers(BattleParams params, boolean amITheAttacker)
+  {
+    if( params.defender.CO == this // We are defending, in a FOREST
+        && (params.combatRef.gameMap.getEnvironment(params.defender.x, params.defender.y).terrainType == TerrainType.FOREST)
+        && (params.defender.model.chassis != UnitModel.ChassisEnum.AIR_HIGH)  // And our unit is actually on the ground.
+        && (params.defender.model.chassis != UnitModel.ChassisEnum.AIR_LOW))
```

Comment:
I think UnitModel.isSurfaceUnit() sounds like a good idea, to centralize this sort of logic.

Reply:
Solid idea.